proposals/dark_mode_extension_icons.md

```diff
@@ -0,0 +1,232 @@
+precedent, followed by the incumbent `icons` key.
+
+The `dark` icon variant can be used by browsers to improve readability of the
+icon. Often this will be when the OS or browser color scheme is dark (often
+referred to as "dark mode") but also when the browser uses a darker shade theme
+which has a background on which the dark icon variant would be more readable.
+The `light` icon variant is used otherwise.
+
+### New Permissions
+
+N/A.
+
+### Manifest File Changes
+
+1. Any icon group that does not contain a "color_scheme" key will apply to all
+available options, e.g. both "light" and "dark".
+
+1. If `icon_variants` contains an icon group with matching conditions, the icon(s)
+specified in the first matching icon group based on insertion order will be used.
+The other icon groups after that match will be ignored.
+   * If there is more than one matching icon object, any without `color_scheme` will
+be applied to every possible color scheme. Therefore, a subsequent matching icon
+object with a `color_scheme` will not be used.
+   * Incorrectly typed `color_scheme` values will be ignored, with an optional warning.
+   For example, `color_scheme: ["unknown"]` will be treated as an empty array.
+   * If only one icon object is defined with a specific color scheme, that
+   icon object will be applied to all color schemes. It will be the icon used.
+
+1. If only one icon group is supplied, all matching conditions will be ignored.
+1. icon_variants will not cause an error in the event that it is invalid
+Instead, only the faulty icon group(s) will be ignored, with an optional
+warning. Warnings are preferred over errors because they're more adaptable to
+changes in the future.
+1. Neither `"dark"` nor `"light"` color schemes are required.
+1. If the top-level `icon_variants` key is provided, the top level `icons` key
+will be ignored.
+   1. For `action` icons, the browser first checks for `icon_variants` in the `action`.
+If not specified, it falls back to `default_icon`, then to the top-level `icon_variants`
+or `icons`.
+   1. Any programmatically set icons using `action.setIcon()` would override the
+declaratively defined icons mentioned above.
+1. The `"16"` is a size in `{"16": "icon.png"}` and any number can be used as a size, per
+the browser’s icon requirements. The word `"any"` can also be used in place of a number
+to represent an icon that can be shown at any size (usually vector images). The icon size
+used by the browser will be determined as follows:
+   1. **Raster images:** Sizes are in pixels. For high-density devices (2x, 3x, etc.), the
+browser looks for double or triple the desired point size (e.g., 32 or 48 pixels for a 16
+point request). If the exact pixel size is unavailable, the next largest pixel size or `"any"`
+will be used.
+   1. **Vector images:** Sizes are in points, ensuring device independence. If the exact
+point size is unavailable, an integer multiple (e.g. 32, 48, etc.) or `"any"` will be used.
+
```

Comment:
```suggestion
`action.setIcon({variants})` will not throw when giving it icon groups with properties it does not understand. Those icon groups will simply be ignored for making matches. If however none of the icon groups are supported, an exception will be thrown allowing both feature detection and specifying fallbacks without requiring feature detection.
```

Reply:
Also hard wrap.

Reply:
This was included manually due to inability to click to commit some suggestions in the UI because of an "outdated" label.